Documentation for SuppressRefreshTokenRotationExecutor (#26664)
keycloak/keycloak: docs/documentation/server_admin/topics/sso-protocols/con-oidc-auth-flows.adoc

See the <<_service_accounts,Service Accounts>> chapter for more information.

[[_refresh_token_grant]]
==== Refresh token grant

> By default, {project_name} return refresh tokens in the token responses from most of the flows. Some exceptions are implicit flow or client credentials grant described above.

Reviewer: ```suggestion
By default, {project_name} returns refresh tokens in the token responses from most of the flows. Some exceptions are implicit flow or client credentials grant described above.
```

> Refresh token is tight to the user session of SSO browser session and can be valid for the whole lifetime of the user session. But note that client should send refresh-token request at least once per

Reviewer: ```suggestion
Refresh token is tied to the user session of the SSO browser session and can be valid for the lifetime of the user session. However, that client should send a refresh-token request at least once per
```

> specified intervals as otherwise the session can be considered "idle" and can expire. See the <<_timeouts,timeouts section>> for more information.

Reviewer: ```suggestion
specified interval. Otherwise, the session can be considered "idle" and can expire. See the <<_timeouts,timeouts section>> for more information.
```

{project_name} supports <<_offline-access,offline tokens>>, which can be used typically when client needs to use refresh token even if corresponding browser SSO session is already expired.

[[_refresh_token_rotation]]
===== Refresh token rotation

> It is possible to specify that refresh token is considered invalid once it is used. This means that client must always save the refresh token from the last refresh response because older refresh tokens,

Reviewer: ```suggestion
It is possible to specify that the refresh token is considered invalid once it is used. This means that client must always save the refresh token from the last refresh response because older refresh tokens,
```

which were already used, would not be considered valid anymore by {project_name}. This is possible to set with the use of _Revoke Refresh token_ option as specified in the <<_timeouts,timeouts section>>.

> {project_name} also supports the possibility when there is no refresh token rotation at all. So refresh token is returned during login, but subsequent responses from refresh-token requests will not

Reviewer: ```suggestion
{project_name} also supports the situation that no refresh token rotation exists. In this case, a refresh token is returned during login, but subsequent responses from refresh-token requests will not
```
